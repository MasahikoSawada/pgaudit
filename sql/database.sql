-- Set up relations for regression test
CREATE USER foo_user;
CREATE USER bar_user;
CREATE DATABASE foo;
CREATE DATABASE bar;

\c foo
CREATE FUNCTION test_func() RETURNS INT AS
$$
BEGIN
    RETURN 0;
END;
$$ LANGUAGE plpgsql;

\c bar
CREATE FUNCTION test_func() RETURNS INT AS
$$
BEGIN
    RETURN 0;
END;
$$ LANGUAGE plpgsql;

-- Test for database rule.
-- SQL on 'foo' database must be logged.

-- logged
\c foo foo_user
SELECT test_func();

-- not logged
\c bar foo_user
SELECT test_func();
