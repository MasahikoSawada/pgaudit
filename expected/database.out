-- Set up relations for regression test
CREATE USER foo_user;
CREATE USER bar_user;
CREATE DATABASE foo;
CREATE DATABASE bar;
\c foo
CREATE FUNCTION test_func() RETURNS INT AS
$$
BEGIN
    RETURN 0;
END;
$$ LANGUAGE plpgsql;
NOTICE:  AUDIT: SESSION,1,1,DDL,CREATE FUNCTION,,"CREATE FUNCTION test_func() RETURNS INT AS
$$
BEGIN
    RETURN 0;
END;
$$ LANGUAGE plpgsql;",<not logged>
\c bar
CREATE FUNCTION test_func() RETURNS INT AS
$$
BEGIN
    RETURN 0;
END;
$$ LANGUAGE plpgsql;
-- Test for database rule.
-- SQL on 'foo' database must be logged.
-- logged
\c foo foo_user
SELECT test_func();
NOTICE:  AUDIT: SESSION,1,1,FUNCTION,EXECUTE,FUNCTION,public.test_funcSELECT test_func();,<not logged>
 test_func 
-----------
         0
(1 row)

-- not logged
\c bar foo_user
SELECT test_func();
 test_func 
-----------
         0
(1 row)

